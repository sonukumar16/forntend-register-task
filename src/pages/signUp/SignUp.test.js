import { mount, shallow } from "enzyme";
import { render, screen, cleanup, waitFor, fireEvent } from "@testing-library/react"
import renderer from 'react-test-renderer';
import {Formik} from "formik"

import SignUp from './';


describe('SignUp test cases', () => {

    let wrapper;
    beforeEach(()=>{
      wrapper = shallow(<SignUp />);
    })
    
    afterEach(cleanup);

    it('should match snapshot', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render button correctly', () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.find('button')).toBeTruthy();
    });

    it('Should have Sign Up text', () => {
        const { getByText } = render(wrapper)
        expect(getByText('Sign Up')).toHaveTextContent("Sign Up");
    });
    
    it('Should submit signup data successfully', async() => {
        const { container, } = render(<SignUp />)
        const fistName = container.querySelector('input[name="firstName"]');
        const lastName = container.querySelector('input[name="lastName"]');
        const email = container.querySelector('input[name="email"]');
        const password = container.querySelector('input[name="password"]');
        const confirmPassword = container.querySelector('input[name="confirmPassword"]');
        const mobile = container.querySelector('input[name="mobile"]');
        const submit = container.querySelector('button[type="submit"]');
        const termsCondition = container.querySelector('input[name="termsCondition"]');

        await waitFor(() => {
            fireEvent.change(fistName, {
              target: {
                value: "mockFirstName"
              }
            })
          })
        await waitFor(() => {
            fireEvent.change(lastName, {
              target: {
                value: "mockLastName"
              }
            })
          })

          await waitFor(() => {
            fireEvent.change(email, {
              target: {
                value: "mock@email.com"
              }
            })
          })
        
          await waitFor(() => {
            fireEvent.change(password, {
              target: {
                value: "password123"
              }
            })
          })

          await waitFor(() => {
            fireEvent.change(confirmPassword, {
              target: {
                value: "password123"
              }
            })
          })

          await waitFor(() => {
            fireEvent.change(mobile, {
              target: {
                value: "+971525343461"
              }
            })
          })

          await waitFor(() => {
            fireEvent.click(termsCondition)
          })
        
          await waitFor(() => {
            fireEvent.submit(submit);
          })

        const expectedData= {
            firstName: "mockFirstName",
            lastName: "mockLastName",
            email: "mock@email.com",
            password: "password123",
            confirmPassword: "password123",
            mobile: "+971525343461",
            termsCondition: true,
        }

        const actualData = {
            firstName: fistName.getAttribute("value"),
            lastName: lastName.getAttribute("value"),
            email: email.getAttribute("value"),
            password: password.getAttribute("value"),
            confirmPassword: confirmPassword.getAttribute("value"),
            mobile: mobile.getAttribute("value"),
            termsCondition:termsCondition.checked,
        }
        expect(actualData).toStrictEqual(expectedData);
    });
});
