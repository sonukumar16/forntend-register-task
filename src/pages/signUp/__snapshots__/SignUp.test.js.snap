// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp test cases should match snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Sign up
    </h1>
    <div
      className="MuiContainer-root py-4 MuiContainer-maxWidthLg"
    >
      <form
        className="makeStyles-form-5"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
              >
                First Name
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="firstName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-9"
                  >
                    <span>
                      First Name
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
              >
                Last Name
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="lastName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-9"
                  >
                    <span>
                      Last Name
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
              >
                Email Address
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-9"
                  >
                    <span>
                      Email Address
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink={false}
              >
                Mobile
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="mobile"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-9"
                  >
                    <span>
                      Mobile
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
              >
                Password
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-9"
                  >
                    <span>
                      Password
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink={false}
              >
                Confirm Password
                <span
                  aria-hidden={true}
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input"
                  disabled={false}
                  name="confirmPassword"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-9"
                  >
                    <span>
                      Confirm Password
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <fieldset
              className="MuiFormControl-root"
            >
              <legend
                className="MuiFormLabel-root"
              >
                Gender
              </legend>
              <div
                className="MuiFormGroup-root makeStyles-group-14"
                defaultValue=""
                form={
                  Object {
                    "dirty": false,
                    "errors": Object {},
                    "getFieldHelpers": [Function],
                    "getFieldMeta": [Function],
                    "getFieldProps": [Function],
                    "handleBlur": [Function],
                    "handleChange": [Function],
                    "handleReset": [Function],
                    "handleSubmit": [Function],
                    "initialErrors": Object {},
                    "initialStatus": undefined,
                    "initialTouched": Object {},
                    "initialValues": Object {
                      "confirmPassword": "",
                      "email": "",
                      "firstName": "",
                      "gender": "",
                      "lastName": "",
                      "password": "",
                      "termsCondition": false,
                    },
                    "isSubmitting": false,
                    "isValid": true,
                    "isValidating": false,
                    "registerField": [Function],
                    "resetForm": [Function],
                    "setErrors": [Function],
                    "setFieldError": [Function],
                    "setFieldTouched": [Function],
                    "setFieldValue": [Function],
                    "setFormikState": [Function],
                    "setStatus": [Function],
                    "setSubmitting": [Function],
                    "setTouched": [Function],
                    "setValues": [Function],
                    "status": undefined,
                    "submitCount": 0,
                    "submitForm": [Function],
                    "touched": Object {},
                    "unregisterField": [Function],
                    "validateField": [Function],
                    "validateForm": [Function],
                    "validateOnBlur": true,
                    "validateOnChange": true,
                    "validateOnMount": false,
                    "values": Object {
                      "confirmPassword": "",
                      "email": "",
                      "firstName": "",
                      "gender": "",
                      "lastName": "",
                      "password": "",
                      "termsCondition": false,
                    },
                  }
                }
                onBlur={[Function]}
                role="radiogroup"
              >
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiRadio-root makeStyles-root-11"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input-18"
                        disabled={false}
                        name="gender"
                        onChange={[Function]}
                        type="radio"
                        value="female"
                      />
                      <span
                        className="makeStyles-icon-12"
                      />
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Female
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiRadio-root makeStyles-root-11"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input-18"
                        disabled={false}
                        name="gender"
                        onChange={[Function]}
                        type="radio"
                        value="male"
                      />
                      <span
                        className="makeStyles-icon-12"
                      />
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Male
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiRadio-root makeStyles-root-11"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input-18"
                        disabled={false}
                        name="gender"
                        onChange={[Function]}
                        type="radio"
                        value="other"
                      />
                      <span
                        className="makeStyles-icon-12"
                      />
                    </span>
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Other
                  </span>
                </label>
              </div>
            </fieldset>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <label
              className="MuiFormControlLabel-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-18"
                    data-indeterminate={false}
                    name="termsCondition"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                I accept terms & condition.
              </span>
            </label>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-6 MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Sign Up
          </span>
        </button>
      </form>
    </div>
  </div>
</main>
`;
